### ISSUE TITLE
Save searches

### ISSUE THREAD

**Issue by CraigClark** [bcgov/MFIN-Data-Catalogue#92]
**OP timer**

https://openplus.monday.com/boards/4092908516/pulses/4733957812

---

## User story

As an authenticated user (not data custodian), I want to view a list of my "Saved searches" so that I can manage them, including deleting them.

## This feature support the following requirements

[alpha release](https://openplus.monday.com/boards/3958997205/pulses/4605939234)

## Additional context

When a user does a search, they need to be able to save the search. 

### save the search block on search page

- Position on the right above the facets
- block title _Save this search_
- field label _Search name_
- text input field is required
- button label _save search_
- confirmation message with the following html    
  ```
  <div class="alert alert-success" role="alert">
    Your saved search was successfully created.
  </div>```

### saved searches on dashboard

We still don't have the UI designed.

- saved searches in a table
- there should be an _Edit_ button that allows, clicking it goes to a page where I can change the label. This page should have a save button and a delete button
- there should be a _Delete_ button to delete the saved search

_Screenshot of save search block_
![](https://github.com/bcgov/MFIN-Data-Catalogue/assets/3910508/cbaccd60-9204-4f20-abee-95e05d93084b)


## Proposed solution

<!-- Any ideas you have on how the feature could be implemented. -->

This is implemented on EVRI, do the same thing.

### button classes

- For save search  button, use `<button type="button" class="btn btn-primary">Save</button>`
- For edit link, use `<a href="#" class="btn btn-primary">Edit</a>`
- For delete link use `<a href="#" class="btn btn-danger">Delete</a>`

## Estimated level of effort

<!-- delete things that don't apply -->

- 1 day

## Definition of done (DoD)
- [x] There is a functioning _save search_ block on the search page
- [x] There is a confirmation message when I save a search
- ~~On my dashboard I see searches I've saved, clicking one of these searches links to the search page with custom facets and search terms applied~~
- ~~on the dashboard I can delete my saved searches~~
- ~~on the dashboard I can edit the label for my saved search~~
- [x] On the dashboard, there is a link "My saved searches" to the saved-searches page provided by the module

## Testing

<!-- You only need one type of test. Functional, site or manual. Please delete the test types you are _not_ using. -->

### Automated functional tests

This test required automated testing. @lkmorlan is the best resource for tthis

- [x] I have written functional tests for this feature
- [x] I have run the functional tests and they pass

### Manual testing

Automated tests test the dashboard. The search page is not tested due to the issue with `bootstrap_layouts`

Manually test that the saved-search block exists.

**Comment by CraigClark** [comment]
@gurjinder12 I tested this locally using your branch.

I got the save search box on the search page, when I save the search I get a confirmation. That is all good.

However, when I go to `/user/1`, I expect to see my saved search on the dashboard. There is nothing there.
I double checked that I did the `cim`, it does import views.

_Dashboard screehshot after saving a search_ I would expect to see a table with my saved search here

![image](https://github.com/bcgov/MFIN-Data-Catalogue/assets/3910508/d7e9509c-1e64-4042-a0b5-8e6debed8bc5)
